### PR TITLE
Extend BuildTime logs and skip analyzers in code gen, code modify phase

### PR DESCRIPTION
**Pull request by gayaldassanayake**
## Purpose
$ subject

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/39488
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/39503

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by Thevakumar-Luheerathan — DISMISSED** [review]

File: project-api/test-artifacts/compiler-plugin-with-analyzer-generator-modifier/src/main/java/io.gayal.plugins.combined/CombinedCompilerPlugin.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        pluginContext.addCodeAnalyzer(new CodeAnalyzer() {
+            @Override
+            public void init(CodeAnalysisContext codeAnalysisContext) {
+                System.out.println("Hello from the analyzer");
+            }
+        });
+        pluginContext.addCodeGenerator(new CodeGenerator() {
+            @Override
+            public void init(CodeGeneratorContext codeGeneratorContext) {
+                System.out.println("Hello from the generator");
```
Comment (azinneera) [inline project-api/test-artifacts/compiler-plugin-with-analyzer-generator-modifier/src/main/java/io.gayal.plugins.combined/CombinedCompilerPlugin.java:39]: Shall we change this to report a diagnostic?

Comment (gayaldassanayake) [reply project-api/test-artifacts/compiler-plugin-with-analyzer-generator-modifier/src/main/java/io.gayal.plugins.combined/CombinedCompilerPlugin.java:39]: `runCodeModifers()`, `runCodeGenerators()` methods ignore the diagnostics from the code analyzing part even when the analyzers are run due to the identified bug. Therefore we do not receive the diagnostics from the code analyzer at the code modifier execution stage anyway. 

Had to use an sout as a result.

Comment (Dilhasha) [reply project-api/test-artifacts/compiler-plugin-with-analyzer-generator-modifier/src/main/java/io.gayal.plugins.combined/CombinedCompilerPlugin.java:39]: I think an sout fails the checkstyles checks. 
May be return the error and report the diagnostic at a different level?

Comment (gayaldassanayake) [reply project-api/test-artifacts/compiler-plugin-with-analyzer-generator-modifier/src/main/java/io.gayal.plugins.combined/CombinedCompilerPlugin.java]: Added the logs to a file and checked

**Review by Dilhasha — APPROVED** [review]

**Review by azinneera — APPROVED** [review]

**Pull request merged**